Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs

```diff
@@ -363,17 +421,48 @@ private void OnBoundUIOpen(EntityUid uid, StorageComponent storageComp, BoundUIO
         }
     }
 
-    private void OnContainerModified(EntityUid uid, StorageComponent component, ContainerModifiedMessage args)
+    private void OnEntInserted(Entity<StorageComponent> entity, ref EntInsertedIntoContainerMessage args)
+    {
+        // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+        if (entity.Comp.Container == null)
+            return;
+
+        if (args.Container.ID != StorageComponent.ContainerId)
+            return;
+
+        if (_net.IsServer)
+        {
```

Comment:
Why is this wrapped in a server check?

Is this supposed to be an ApplyingState check or something?